stories/Folder-Structure.stories.mdx

```diff
@@ -0,0 +1,44 @@
+import { Meta } from '@storybook/addon-docs';
+import { H1 } from '@leafygreen-ui/typography';
+import BannerImage from '../.storybook/static/leafygreen-ui-banner.png';
+
+<Meta title="Developer Guide/Package Folder Structure" />
+
+### Folder Structure
+
+- `index.tsx`
+- `styles.ts`
+- `types.ts`
+- `<ComponentName>.tsx`
+- `<ComponentName>.spec.tsx**`
+- `<ComponentName>.stories.tsx**`
+- `any subcomponent directories`
+
+***only if the stories or tests pertain specifically to the component in this directory; meaning it may not be necessary for many internal subcomponents*
+
+The folder structure should be identical for all components - whether they are considered components that are exported to npm, or subcomponents contained within those packages. In other words, **all subcomponents are simply components**, and they should be structured identically to reflect that.
+
+Even if a package only has one component, it should have a subdirectory. For example, the path to the component file for `Badge` would be `/badge/src/Badge/Badge.tsx`. While this could be seen as clutter, it allows us to differentiate code that is relevant to the entire *package* from code that only pertains to the *component*. It is also more scalable as adding subcomponents in the future would simply involve adding a new directory, instead of having to move all files in the package.
```

Comment:
I would disagree with this point. I would prefer all base component files to be in the root of `src` than to create an additional level of directory nesting. 

Most components won't have subcomponents, which makes this needlessly deep, and even for components with subcomponents, I don't see any issue with having the main component's files being in the src root.

i.e.
```
button/src
 - index.tsx
 - Button.tsx
 - Button.styles.tsx
 - Button.types.tsx
 - Button.stories.tsx
 - Button.spec.tsx
 - ButtonIcon
 - - index.tsx
 - - ButtonIcon.tsx
 - - etc...
```

Reply:
I do think that this makes a component needlessly deep but I like the consistency of keeping everything in a subcomponent. I think the code above would look a bit neater if everything for `Button` was in its own folder.

Reply:
It may be a bit cleaner, but IMO it makes it harder to parse. If we have several folders within a `src` folder it makes it hard to parse which is the _main_ component, and which are subcomponents. Take the `SearchInput` component for instance. It takes a bit more mental effort to parse which is the main component, vs a case where the main component files are in the root of `src` 

```
search-input/src/
 - SearchInput
 - SearchResult
 - SearchResultMenu
 - SearchResultGroup
```